### PR TITLE
Use huggingface datasets repo instead of data in spaces for HF demos.

### PR DESCRIPTION
**Pull request by nsthorat**
See "datasets" here for uploaded datasets: https://huggingface.co/lilacai

### CODE REVIEW

File: docker_start.py [hunk 1]
```diff
@@ -41,21 +52,30 @@ def main() -> None:
     return
 
   delete_old_files()
+
+  print('readme:', os.path.abspath('README.md'))
```
Comment (dsmilkov) [inline docker_start.py:56]: do you need this and the next print?

File: docker_start.py [hunk 2]
```diff
@@ -41,21 +52,30 @@ def main() -> None:
...
+    snapshot_download(
+      repo_id=lilac_hf_dataset,
+      repo_type='dataset',
+      token=env('HF_ACCESS_TOKEN'),
+      local_dir=get_datasets_dir(data_path()),
+      ignore_patterns=['.gitattributes', 'README.md'])
+
   snapshot_dir = snapshot_download(repo_id=repo_id, repo_type='space', token=env('HF_ACCESS_TOKEN'))
-  # Copy datasets.
+  # # Copy datasets.
```
Comment (dsmilkov) [inline docker_start.py:77]: double comment

File: scripts/deploy_demo.py [hunk 3]
```diff
@@ -64,18 +66,43 @@
   type=bool,
   is_flag=True,
   default=False)
+@click.option(
+  '--make_datasets_public',
+  help='When true, sets the huggingface datasets uploaded to blic. Defaults to false.',
```
Comment (dsmilkov) [inline scripts/deploy_demo.py:71]: typo: blic --> public

File: scripts/deploy_demo.py [hunk 4]
```diff
@@ -64,18 +66,43 @@
...
+
+      print(f'Downloading dataset from HuggingFace "{repo_id}": ', dataset)
+      snapshot_download(
+        repo_id=repo_id,
+        repo_type='dataset',
+        token=env('HF_ACCESS_TOKEN'),
+        local_dir=get_datasets_dir(data_dir),
+        ignore_patterns=['.gitattributes', 'README.md'])
+
+    # huggingface_hub.snapshot_download(
```
Comment (dsmilkov) [inline scripts/deploy_demo.py:101]: cleanup

**Review by dsmilkov — APPROVED** [review]

File: docker_start.py (hunk 1 above)
Comment (nsthorat) [reply docker_start.py]: nope, done

File: docker_start.py (hunk 2 above)
Comment (nsthorat) [reply docker_start.py]: done

File: scripts/deploy_demo.py (hunk 4 above)
Comment (nsthorat) [reply scripts/deploy_demo.py]: done

**Pull request merged**